dashboard/app/models/levels/level_group.rb

```diff
@@ -89,13 +89,53 @@ def pages
     end
   end
 
+  def assign_attributes(params)
+    @pages = nil
+    super(params)
+  end
+
   def plc_evaluation?
     levels.map(&:class).uniq == [EvaluationMulti]
   end
 
   # Surveys: How many students must complete a survey before any results are shown.
   SURVEY_REQUIRED_SUBMISSION_COUNT = 5
 
+  # Perform a deep copy of this level by cloning all of the levels within it
+  # using the same suffix, and write them to the new level definition file.
+  def clone_with_suffix(new_suffix)
+    level = super(new_suffix)
+    level.add_suffix_to_grouped_levels(new_suffix)
+    level.rewrite_dsl_file(LevelGroupDSL.serialize(level))
+    level
+  end
+
+  # Clone of the levels belonging to this level, adding the specified suffix to
+  # the level name.
+  def add_suffix_to_grouped_levels(new_suffix)
```

Comment:
Nit: I would have called this `clone_grouped_levels` or `clone_grouped_levels_with_suffix`.

Reply:
Good call. It looks like these are called "sublevels" rather than "grouped levels", so I'm changing the wording to "sublevels" too.